test/functional/feature_txindex_compatibility.py

```diff
@@ -55,23 +54,13 @@ def run_test(self):
         drop_index_chain_dir = self.nodes[1].chain_path
         shutil.rmtree(drop_index_chain_dir)
         shutil.copytree(legacy_chain_dir, drop_index_chain_dir)
-        self.nodes[1].assert_start_raises_init_error(
-            extra_args=["-txindex"],
-            expected_msg="Error: The block index db contains a legacy 'txindex'. To clear the occupied disk space, run a full -reindex, otherwise ignore this error. This error message will not be displayed again.",
-        )
         # Build txindex from scratch and check there is no error this time
         self.start_node(1, extra_args=["-txindex"])
```

Comment:
Could add these lines to check that if user is upgrading from an older node, they'll still get a helpful message on how to use txindex again (albeit without cleaning up unused space)?
```suggestion
        self.log.info("Check that \"Use -txindex\" error is thrown and -txindex works normally")
        self.start_node(1)
        assert_raises_rpc_error(-5, "Use -txindex", lambda: self.nodes[1].getrawtransaction(txid=spend_utxo["txid"]))
        self.restart_node(1, extra_args=["-txindex"])
```

Reply:
Yeah you're right, this doesn't add anything. I wanted to check that a user had clear instructions on how to build the new txindex, but they wouldn't have to do anything at all, as they're already expected to keep `-txindex` in startup options every time.

Could explicitly add a check that we're indeed building a new txindex (as opposed to somehow using the legacy one), but that doesn't seem like a likely issue either so may not be worth testing? No strong view, so can be marked as resolved.

```suggestion
        with self.nodes[1].assert_debug_log(expected_msgs=['Syncing txindex with block chain from height 0']):
            self.start_node(1, extra_args=["-txindex"])
```